### ISSUE TITLE
A puzzler: Google Sheet errors while same spreadsheet saved from within Excel is fine

### ISSUE THREAD

**Issue by jsvine** [saulpw/visidata#103]
## Specs

- VisiData v0.99
- Python 3.6.0
- MacOS
- Terminal.app

## Observed Behavior

This one has me scratching my head a bit. It seems that some spreadsheets, when downloaded from Google Sheets, cause `vd` to throw the following error when trying to open any sheet (though not the workbook itself):

```
Traceback (most recent call last):
  File "[[redacted]]/python3.6/site-packages/visidata/vdtui.py", line 554, in toplevelTryFunc
    ret = func(*args, **kwargs)
  File "[[redacted]]/python3.6/site-packages/visidata/loaders/xlsx.py", line 29, in reload
    for i in range(worksheet.max_column):
TypeError: 'NoneType' object cannot be interpreted as an integer
```

... *but* when you open the spreadsheet in Excel and resave it, `vd` is then able to open the sheet with no problem.

At first, I thought it was a problem *just* with `openpyxl` and its `.max_column` sheet property but, as you'll see in the example below, that doesn't seem to explain it. Not being super familiar with VisiData's integration of `openpyxl`, I'm a bit stumped.

## Example

Here is a exceedingly simple spreadsheet I created in Google Sheets — no special formatting, formulas, or anything like that: 
[test-google-sheets.xlsx](https://github.com/saulpw/visidata/files/1613613/test-google-sheets.xlsx)

When I run `vd test-google-sheets.xlsx` and then press `Enter` on "Sheet 1", I get the error described above.

Here is the very same spreadsheet, which I've then opened in Excel and resaved. (Just a simple `Command-s`, no special settings.): 
[test-resaved-from-excel.xlsx](https://github.com/saulpw/visidata/files/1613614/test-resaved-from-excel.xlsx)

When I run `vd test-resaved-from-excel.xlsx` and then press `Enter` on "Sheet 1", I see the spreadsheet — no error.

As a simple test, I wrote this Python script, which I saved as `get_maxcol.py`:

```python
#!/usr/bin/env python
from openpyxl import load_workbook
import sys
wb = load_workbook(sys.stdin)
print(wb._sheets[0].max_column)
```

If the problem were with `openpyxl` alone, I'd expect the result of running `python get_maxcol.py <  test-google-sheets.xlsx` to be `None`. But it's not; it returns `2` — just like the other file.

Perhaps I'm overlooking an obvious answer?

**Comment by saulpw** [comment]
The difference is the [`read_only=True` passed to `load_workbook`](https://github.com/saulpw/visidata/blob/stable/visidata/loaders/xlsx.py#L17) in VisiData.  This uses an optimized loader, which takes some shortcuts so that it can read the file faster.  (I've tried removing it in the past, so that the openpyxl sheet could be edited in-place, but that slows loading down by 2-3x).

According to [this openpyxl issue](https://bitbucket.org/openpyxl/openpyxl/issues/278/get_highest_row-column-are-unreliable), if the sheet doesn't contain the dimensions, the only way to get them is to iterate through the whole sheet (which can be quite expensive).

I've rejiggered the xlsx loader to be a little smarter, so at least it succeeds in such a case.  At worst, sheets without dimensions will lack a working progress indicator while loading.

I'm not sure if we should try to take this for 1.0.
